src/flow.ts

```diff
@@ -200,6 +206,8 @@ export class Flow {
   scopedLocals: Map<string,Local> | null = null;
   /** Local flags. */
   localFlags: LocalFlags[];
+  /** Field flags. */
+  fieldFlags: Map<string, FieldFlags>;
```

Comment:
Makes me wonder if each field should instead have an incrementing unique id (per class, in layout order) so we can use an array here.

Reply:
Another optimization might be to use a `null` map anywhere but in constructors. Will save quite a bit of memory and reduce GC pressure because there can be quite a lot of flows.

Reply:
> Makes me wonder if each field should instead have an incrementing unique id (per class, in layout order) so we can use an array here.

Right, this makes sense, it was my original approach – similar to what we do with locals –  it's just that I couldn't find a straightforward way to go about it for assignments since I don't have evident access to the field index but I guess we could list fields by index as well. 

I'll try that out!